pkg/cmd/cli/cmd/project.go

```diff
@@ -43,71 +52,77 @@ func NewCmdProject(f *clientcmd.Factory, out io.Writer) *cobra.Command {
 				if len(currentProject) > 0 {
 					_, err := oClient.Projects().Get(currentProject)
 					if err != nil {
-						if errors.IsNotFound(err) {
+						if errors.IsNotFound(err) || clientcmd.IsForbidden(err) {
 							glog.Fatalf("The project '%v' specified in your config does not exist or you do not have rights to view it.", currentProject)
 						}
 						checkErr(err)
 					}
 
-					fmt.Printf("Using project '%v'.\n", currentProject)
+					fmt.Printf("Using context '%v' with project '%v'.\n", rawCfg.CurrentContext, currentProject)
```

Comment:
Wasn't sure what your comment meant, but make this an if block "if currentProject != currentContext { Using project %v from context named %v }   else { Using project %v }"